src/Sparrow/ByteString.cs

```diff
@@ -735,9 +735,6 @@ public void Reset()
             if (_disposed)
                 ThrowObjectDisposed();
 
-            if (_wholeSegments.Count == 2)
-                return; // nothing to do
-
             Array.Clear(_internalReusableStringPoolCount, 0, _internalReusableStringPoolCount.Length);
```

Comment:
This is the actual issue, we are not resetting the pointers. I wouldnt though, perform both changes simultaneously (fix + size change) because you can be impacting performance hard... Fix the issue where you reset the actual pointers, check if the problem continue to be repro and if that is the case then perform the size change.

Reply:
We cannot reset pointers since we have places where we want to keep pointers until a call to Reset().
For example [here](https://github.com/ravendb/ravendb/blob/v4.1/src/Voron/Data/Tables/TableSchema.cs#L366) - we want to keep the Slice alive until the transaction commits (and the context is reset when it is returned to the pool)